core/src/datayoga_core/step.py

```diff
@@ -63,6 +64,10 @@ async def run(self, worker_id: int):
             try:
                 processed_entries, filtered_entries, rejected_entries = await self.block.run(entry)
 
+                prom.processed_entries.labels(step=self.id).inc(len(processed_entries))
+                prom.filtered_records.labels(step=self.id).inc(len(filtered_entries))
+                prom.rejected_records.labels(step=self.id).inc(len(rejected_entries))
+
```

Comment:
again, does it make sense to do it even if Prometheus is not started?

Reply:
It just increment a value... I don't think we're buying time here if we add condition.